### PR TITLE
[webview_flutter_wkwebview][webview_flutter_android] Fixes bug where the `WebView`s could not be released

### PR DESCRIPTION
**Pull request by bparrishMines**
Fixes https://github.com/flutter/flutter/issues/118579

tldr;
Fixed a bug caused by misusing `withWeakRefenceTo`. And fixed integration tests that started to stall because of the change.

So, apparently this causes a circular reference:

```dart
late final WKWebView _webView = withWeakRefenceTo(this, (
    WeakReference<WebKitWebViewController> weakReference,
  ) {
    return _webKitParams.webKitProxy.createWebView(
      _webKitParams._configuration,
      observeValue: (
        String keyPath,
        NSObject object,
        Map<NSKeyValueChangeKey, Object?> change,
      ) {
        // use weakReference
        }
      },
    );
  });
```

but this doesn't:

```dart
late final WKWebView _webView = _webKitParams.webKitProxy.createWebView(
    _webKitParams._configuration,
    observeValue: withWeakRefenceTo(this, (
      WeakReference<WebKitWebViewController> weakReference,
    ) {
      return (
        String keyPath,
        NSObject object,
        Map<NSKeyValueChangeKey, Object?> change,
      ) {
        // use weakReference
      };
    }),
    instanceManager: _webKitParams._instanceManager,
  );
```

It seems important that the callback method can't include any references to external values. I should have followed my own [example](https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter_wkwebview/lib/src/common/weak_reference_utils.dart#L12) 🤦‍♂️.

It also seems this change affected a bunch of integration tests.  My assumption that Android required displaying the WebView to get the `onPageFinished` callback may have been incorrect. The callback doesn't work consistently unless a widget is loaded. This now happens on iOS AND Android. I was unable to debug why this is the case. It could just be quirk of integration tests. The callbacks still work in the example app when running in debug and release mode, so I don't think it is indicative of another bug.

------------------------

Also tack on an additional quick fix for Android:

Fixes https://github.com/flutter/flutter/issues/119332

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by arifje** [comment]
When will this fix be merged? Need it to get my project working 😁

File: packages/webview_flutter/webview_flutter_android/lib/src/android_webview_controller.dart [hunk 1]
```diff
@@ -126,10 +128,7 @@ class AndroidWebViewController extends PlatformWebViewController {
   final Map<String, AndroidJavaScriptChannelParams> _javaScriptChannelParams =
       <String, AndroidJavaScriptChannelParams>{};
 
-  // The keeps a reference to the current NavigationDelegate so that the
-  // callback methods remain reachable.
-  // ignore: unused_field
-  late AndroidNavigationDelegate _currentNavigationDelegate;
+  AndroidNavigationDelegate? _currentNavigationDelegate;
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_android/lib/src/android_webview_controller.dart:131]: Making this nullable fixes https://github.com/flutter/flutter/issues/119332

**Review by cyanglaz — APPROVED** [review]
LGTM
